mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -3707,14 +3711,41 @@ async fn check_balance_for_maker_orders(ctx: MmArc, ordermatch_ctx: &OrdermatchC
         // This checks that the order hasn't been removed by another process
         if removed_order_mutex.is_some() {
             maker_order_cancelled_p2p_notify(&ctx, &order);
-            delete_my_maker_order(ctx.clone(), order.clone(), reason)
-                .compat()
-                .await
-                .ok();
+            delete_my_maker_order(ctx.clone(), order.clone(), reason).await;
         }
     }
 }
 
+/// Cancel maker order if coin is offline and return true if cancelled.
+async fn cancel_maker_order_if_coin_offline(
+    ctx: &MmArc,
+    ordermatch_ctx: &OrdermatchContext,
+    order: &MakerOrder,
+) -> bool {
+    if let Ok(Some(coin)) = lp_coinfind(ctx, &order.base).await {
```

Comment:
i think we should also cancel the order if it's not found (tho we also handle it else where i presume). and also cancel it if `lp_coinfind` errors (just for code simplicity, since we know it shouldn't really error).

Reply:
hmmm. I believe lp_ordermatch loop will wait till a coin is active before matching etc. 

> and also cancel it if lp_coinfind errors (just for code simplicity, since we know it shouldn't really error).

And I do not think we want to do this since coin not found seems very different from when it is offline.